### PR TITLE
bpo-46482: add a test for `typing.Annotation.__new__`

### PR DESCRIPTION
**Pull request by sobolevn**
https://bugs.python.org/issue46482

https://bugs.python.org/issue46482

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -4595,6 +4595,13 @@ class A(typing.Match):
 
 class AnnotatedTests(BaseTestCase):
 
+    def test_new(self):
+        with self.assertRaisesRegex(
+            TypeError,
+            'Type Annotated cannot be instantiated',
```
Comment (JelleZijlstra) [inline Lib/test/test_typing.py:4601]: "Type Annotated" is odd wording, it would be clearer as just "Annotated cannot be instantiated".

Comment (sobolevn) [reply Lib/test/test_typing.py:4601]: I have just copy pasted the existing message. I guess it is out of scope 🙂 
Should we open a new issue?

**Comment by sobolevn** [comment]
@Fidget-Spinner any chance to get this merged, please? 🙂

**Comment by Fidget-Spinner** [comment]
@sobolevn sorry but I can't, see https://mail.python.org/archives/list/python-dev@python.org/thread/HU2LH5MX44QZXZ36YS4SOG4ZUPT4E7QT/ for the reason :).

**Comment by sobolevn** [comment]
Oh, this is unexpected! Thanks for letting me know 👍

**Pull request merged**